src/main/kotlin/com/stsf/globalbackend/controllers/HomeworkBallController.kt

```diff
@@ -0,0 +1,62 @@
+package com.stsf.globalbackend.controllers
+
+import com.stsf.globalbackend.exceptions.InsufficientPermissionsException
+import com.stsf.globalbackend.models.HomeworkBall
+import com.stsf.globalbackend.repositories.HomeworkBallRepository
+import com.stsf.globalbackend.repositories.HomeworkRepository
+import com.stsf.globalbackend.request.GenericResponse
+import com.stsf.globalbackend.services.AuthenticationService
+import com.stsf.globalbackend.services.HomeworkBallService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.*
+
+
+// Maybe work on the naming
+@RestController
+@RequestMapping("/api/homework/balls")
+class HomeworkBallController (
+    @Autowired
+    private val auth: AuthenticationService,
+    @Autowired
+    private val ballService: HomeworkBallService,
+    ) {
+
+    @PutMapping("/")
+    fun addBall(@RequestHeader token: String, @RequestParam homeworkId: Long): GenericResponse<HomeworkBall> {
+        val authenticatedUser = auth.getUserByToken(token)
+        return GenericResponse(ballService.addBall(authenticatedUser.id, homeworkId))
+
+    }
+
+    @DeleteMapping("/")
+    fun deleteBall(@RequestHeader token: String, @RequestParam ballId: Long) {
+        val authenticatedUser = auth.getUserByToken(token)
+        val balls = ballService.getAllBallsByUserId(authenticatedUser.id)
+
+        for (ball in balls) {
+            if (ball.id == ballId) {
+                ballService.deleteBallById(ball.id)
+            }
+        }
+
+    }
+
+    @DeleteMapping("/homework")
+    fun deleteBallByHomeworkId(@RequestHeader token: String, @RequestParam homeworkId: Long) {
+        val authenticatedUser = auth.getUserByToken(token)
+        val balls = ballService.getAllBallsByUserId(authenticatedUser.id)
```

Comment:
Same thing here, you can fist fetch the ball by homeworkId and then only check if it's in this list, no need for for loops.

Reply:
Nvm, it can stay like this, just fix the issue above.